pallets/liquid-staking-two/src/lib.rs

```diff
@@ -0,0 +1,607 @@
+    ) {
+        let pool_ledger_per_era = MatchingPoolByEra::<T>::get(&era_index);
+        let success_operation = pool_ledger_per_era.operation_type.and_then(|t| {
+            let operation = StakingOperationHistory::<T>::get(&era_index, &t)?;
+            if operation.status == ResponseStatus::Successed {
+                return Some(t);
+            }
+            None
+        });
+
+        if let Some(staking_operation_type) = success_operation {
+            let current_era = Self::current_era();
+            let (claim_unstake_each_era, claim_stake_each_era) = match staking_operation_type {
+                StakingOperationType::Bond => Self::claim_in_bond_operation(
+                    who,
+                    &era_index,
+                    &current_era,
+                    &user_ledger_per_era,
+                    &pool_ledger_per_era,
+                ),
+                StakingOperationType::Unbond => Self::claim_in_unbond_operation(
+                    who,
+                    &era_index,
+                    &current_era,
+                    &user_ledger_per_era,
+                    &pool_ledger_per_era,
+                ),
+                StakingOperationType::Matching => user_ledger_per_era.remaining_withdrawal_limit(), //if matching, can claim all directly
+                _ => (0, 0),
+            };
+
+            MatchingQueueByUser::<T>::mutate(who, &era_index, |b| {
+                b.claimed_unstake_amount = b
+                    .claimed_unstake_amount
+                    .saturating_add(claim_unstake_each_era);
+                b.claimed_stake_amount =
+                    b.claimed_stake_amount.saturating_add(claim_stake_each_era);
+
+                if b.total_stake_amount == b.claimed_stake_amount
+                    && b.total_unstake_amount == b.claimed_unstake_amount
+                {
+                    // user have already claimed all he can claim in this era, remove it from MatchingQueue
+                    remove_record_from_user_queue.push(era_index);
+                }
+            });
+
+            *withdrawable_unstake_amount =
+                withdrawable_unstake_amount.saturating_add(claim_unstake_each_era);
+            *withdrawable_stake_amount =
+                withdrawable_stake_amount.saturating_add(claim_stake_each_era);
+        }
+    }
+
+    // if bond, need to wait 1 era or get the matching part instantly
+    // users who unstake can get KSM back directly, because this era is about bond operation,
+    // so all user who unstake in this era, no need wait 28 eras.
+    // after waiting 1 era, users who stake get the left part
+    fn claim_in_bond_operation(
+        who: &T::AccountId,
+        claim_era: &EraIndex,
+        current_era: &EraIndex,
+        user_ledger_per_era: &UserLedgerPerEra,
+        pool_ledger_per_era: &PoolLedgerPerEra,
+    ) -> WithdrawalAmount {
+        if claim_era.clone() + T::StakingDuration::get() > current_era.clone() {
```

Comment:
`EraIndex` should be passed because `EraIndex` implemented Copy and it's cheap, we don't need to clone manually